File: leetcode-all/zihui/20210219_1249_Minimum_Remove_to_Make_Valid_Parentheses.java
class Solution {
    public String minRemoveToMakeValid(String s) {
        Stack<Integer> stack = new Stack();
        String[] array = s.split("");
        for (int i = 0; i < array.length; i++) {
            if (array[i].equals("(")) {
                stack.push(i);
            }
            if (array[i].equals(")")) {
                if (stack.empty() == false) {
                    stack.pop();
                } else {
                    array[i] = "";
                }
            }
        }

        while (stack.empty() == false) {
            array[stack.pop()] = "";
        }
        return String.join("", array);
    }
}

File: leetcode-all/zihui/20210219_1249_Minimum_Remove_to_Make_Valid_Parentheses.py
class Solution:
    def minRemoveToMakeValid(self, s: str) -> str:
        stack = []
        s = list(s)
        for idx, val in enumerate(s):
            if val == '(':
                stack.append(idx)
            elif val == ')':
                if stack:
                    stack.pop()
                else:
                    s[idx] = ''
        while stack:
            s[stack.pop()] = ''
        return ''.join(s)

